iac-extensions/terraform/src/main/java/org/sonar/iac/terraform/checks/azure/AzureIpRestrictedAdminAccessCheckPart.java

```diff
@@ -0,0 +1,133 @@
+
+  private static final Pattern PORT_RANGE_PATTERN = Pattern.compile("^(?<from>\\d+)-(?<to>\\d+)$");
+  private static final Set<String> SENSITIVE_PORTS = Set.of("*", String.valueOf(SSH_PORT), String.valueOf(RDP_PORT));
+  private static final Set<String> SENSITIVE_PREFIXES = Set.of("*", "0.0.0.0/0", "::/0");
+
+  @Override
+  protected void registerResourceChecks() {
+    register(AzureIpRestrictedAdminAccessCheckPart::checkNetworkSecurityGroup, "azurerm_network_security_group");
+  }
+
+  public static void checkNetworkSecurityGroup(CheckContext ctx, BlockTree resource) {
+    PropertyUtils.getAll(resource, "security_rule", BlockTree.class).stream()
+      .filter(AzureIpRestrictedAdminAccessCheckPart::conditionsApply)
+      .forEach(rule -> checkSecurityRule(ctx, rule));
+  }
+
+  /**
+   * Check if all attributes beside the port range and ip attribute match the condition to make the rule sensitive
+   */
+  private static boolean conditionsApply(BlockTree rule) {
+    return hasAttributeWithMatchingValue(rule,"direction", "Inbound"::equals)
+      && hasAttributeWithMatchingValue(rule, "access", "Allow"::equals)
+      && hasAttributeWithMatchingValue(rule, "protocol", p -> "Tcp".equals(p) || "*".equals(p));
+  }
+
+  private static void checkSecurityRule(CheckContext ctx, BlockTree rule) {
+    sensitiveDestinationPortRange(rule).ifPresent(sensitivePort ->
+      sensitiveSourcePrefix(rule).ifPresent(sensitivePrefix ->
+        ctx.reportIssue(sensitivePrefix, MESSAGE, new SecondaryLocation(sensitivePort, SECONDARY_MSG))));
+  }
+
+  private static Optional<ExpressionTree> sensitiveDestinationPortRange(BlockTree rule) {
+    Predicate<ExpressionTree> rangeContainsSensitivePort = range -> TextUtils.getValue(range)
+      .filter(AzureIpRestrictedAdminAccessCheckPart::rangeContainsSensitivePort).isPresent();
+
+    return PropertyUtils.get(rule, "destination_port_range", AttributeTree.class)
+      .map(AttributeTree::value)
+      .filter(rangeContainsSensitivePort)
+      .or(() -> sensitiveExpressionInAttributeTuple(rule, "destination_port_ranges", rangeContainsSensitivePort));
+  }
+
+  private static Optional<ExpressionTree> sensitiveSourcePrefix(BlockTree rule) {
+    Predicate<ExpressionTree> isSensitivePrefix = prefixExpression -> TextUtils.matchesValue(prefixExpression, SENSITIVE_PREFIXES::contains).isTrue();
+
+    return PropertyUtils.get(rule, "source_address_prefix", AttributeTree.class)
+      .map(AttributeTree::value)
+      .filter(isSensitivePrefix)
+      .or(() -> sensitiveExpressionInAttributeTuple(rule, "source_address_prefixes", isSensitivePrefix));
+  }
+
+  /**
+   * If the attribute in the block exists and the attribute's value is a tuple,
+   * return the first expression in the tuple elements which matches the sensitive condition
+   */
+  private static Optional<ExpressionTree> sensitiveExpressionInAttributeTuple(BlockTree block, String attribute, Predicate<ExpressionTree> predicate) {
```

Comment:
Minor: the name of this method could be simplified, it doesn't have to do with "sensitive".